search/task_helpers.py

```diff
@@ -0,0 +1,70 @@
+from functools import wraps
+
+from channels.constants import (
+    POST_TYPE,
+)
+from search.indexing_api import gen_post_id
+from search.tasks import (
+    create_document,
+    partially_update_document,
+)
+
+
+def reddit_object_indexer(indexing_func=None):
+    def api_indexing_listener_inner(func):
+        @wraps(func)
+        def wrapped_api_func(*args, **kwargs):
+            reddit_obj = func(*args, **kwargs)
+            indexing_func(reddit_obj)
```

Comment:
We may want to `try...except` + `log.exception` this on any errors so an indexing error doesn't result in a 500 error back to the end user.